pkg/server/deploy/handlers.go

```diff
@@ -0,0 +1,100 @@
+package deploy
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"github.com/luizalabs/teresa-api/models/storage"
+	dpb "github.com/luizalabs/teresa-api/pkg/protobuf/deploy"
+)
+
+const (
+	keepAliveMessage = "\u200B" // Zero width space
+)
+
+type Options struct {
+	KeepAliveTimeout     time.Duration `split_words:"true" default:"30s"`
+	RevisionHistoryLimit int           `split_words:"true" default:"5"`
+}
+
+type Service struct {
+	ops     Operations
+	options *Options
+}
+
+func (s *Service) Make(stream dpb.Deploy_MakeServer) error {
+	var appName, description string
+	content := new(bytes.Buffer)
```

Comment:
Maybe we should a use a temporary file for the content because these tar files are big and keeping them in memory may increase the memory requirements.